libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -890,14 +896,13 @@ GCS_MAVLINK::update(run_cli_fn run_cli, uint32_t max_time_us)
         }
     }
 
-    if (!waypoint_receiving) {
-        hal.util->perf_end(_perf_update);    
-        return;
-    }
-
     uint32_t tnow = AP_HAL::millis();
     uint32_t wp_recv_time = 1000U + (stream_slowdown*20);
 
+    if (!waypoint_receiving) {
+        goto end;
+    }
```

Comment:
Nitpicking: don't change the position of this, it will go to the previous position in a later commit.